Ensure the right values are set in SeriesGroupBy.nunique

aiguofer opened the pull request:
We only need to use the group boundaries as the index for `res` so that
the dimensions match those of `out`. Fixes #13453

 - [x] closes #13453
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [ ] whatsnew entry

Commit:
Ensure the right values are set in SeriesGroupBy.nunique

We only need to use the group boundaries as the index for `res` so that
the dimensions match those of `out`. Fixes #13453

jreback commented:
can you add all 4 comparisions (e.g. ``.count`` == ``.nunique``, https://github.com/pandas-dev/pandas/issues/13453#issue-160499155) and using ``.agg`` https://github.com/pandas-dev/pandas/issues/13453#issuecomment-226295096

so you have 4 cases that you are comparing for equality (you can call them result1-4 and compare)

jreback commented:
pls also add a bug fix note in 0.20.0. lgtm otherwise. ping on green! (with test additions)

jreback commented:
@aiguofer you shouldn't have *any* tests fail. Note we have recently switched to ``pytest``.

Commit:
Add test for #13453 in test_resample and add note to whatsnew

aiguofer commented:
Added test to `resample` since it made more sense there... I also added `.agg('nunique')` for completeness.

When I run tests with `pytest` all passes :)

aiguofer commented:
@jreback all looks good :)

jreback commented:
thanks @aiguofer 

keep em coming!
